docs/advanced/full-node.md

```diff
@@ -0,0 +1,63 @@
+
+- Access to the settings of your router (to allow outbound connections)
+- A safe place to leave your device running
+- If you need a silent node (hard disk drives can be loud), a SSD or USB drive
+
+## Guides
+
+Below are some guides from the community that walk you step-by-step through setting up a full node on various hardware and OSes. For generic high-level instructions, see the [Steps](#steps) section below. 
+
+- [Running a Decred Raspberry Pi Node](https://medium.com/@_Checkmatey_/running-a-decred-raspberry-pi-node-ac605b70c652): A more beginner friedly guide that uses a VNC server (remote desktop). This guide is intented to be a gentle introduction to using SSH on the command line. 
+- [My experience setting up a Decred full-node](https://medium.com/crypto-rocket-blog/my-experience-setting-up-a-decred-full-node-8a9bbf55bc30): Another beginner-friendly guide that walks through setting up a full node on a Raspberry Pi using an Ubuntu terminal on a Windows PC.
+- [Installing dcrd](https://stakey.club/en/installing-dcrd/): A more advanced guide to installing a full node (Mac or Debian) that also covers manual installation (from binaries or source) and configuring dcrd to run on Tor for extra security.
+
+## Steps
+
+Below are the basic high-level steps to install a full node on any hardware and OS. These instructions use [dcrinstall](../wallets/cli/cli-installation.md), the automatic installer and upgrader software. In addition to installing `dcrd`, it will also install `dcrwallet` (which allows you to create wallets) and `dcrctl` (which allows you to control `dcrwallet` and `dcrd` from the command line). You can also just install `dcrd` on its own from [binaries or source files](https://github.com/decred/dcrd#installing-and-updating); a full node does not require a wallet unless you want to use your full node to validate and broadcast your own transactions.
+
+1. **Download and install dcrd**
+	- Follow the instructions for your OS in the [CLI Installation guide](https://docs.decred.org/wallets/cli/cli-installation/). The `dcrd` executable will be installed in a directory named `/decred` under your OS's home directory. 
+2. **Start `dcrd`**
+	- Navigate to the `/decred` directory and launch `dcrd`. See the [Operating System Differences](../wallets/cli/os-differences.md) page for OS-specific commands.
+	- `dcrd` will boot up, begin connecting to peers and downloading the full Decred blockchain.
+3. **(Optional) Create wallet**
+	- If you would like to use your full node to validate and broadcast your wallet's transactions, you'll need to create and set up a wallet. See the [dcrwallet Setup Guide](../wallets/cli/dcrwallet-setup.md) for instructions.
+4. **Enable incoming connections**
+	- `dcrd` will automatically begin downloading the blockchain and connecting to peers. However, if you want to help the network and have your node start validating transactions, you'll need to configure your internet router to allow inbound connections. 
```

Comment:
Right, but how would it get new transactions to validate if it isn't accepting inbound connections?

Reply:
Inbound or outbound only describes how a connection is established, it does not describe the flow of information. Broadly speaking, an outbound connection is one you have initiated yourself by dialing out to a peer, and an inbound connection is coming in to you from another peer. 

Once a connection is established (either inbound or outbound) it enables bidirectional communication. This is how dcrd is able to participate and fully validate, even if it only has outbound peers. Inbound peers aren't strictly necessary at all, but for a node to actually have a beneficial effect on the network you should really let them in.

Reply:
Have the below now. Though wondering, do we want to recommend inbound peers also, for the benefits they provide the network?

1. **Enable incoming connections**
	- `dcrd` will automatically begin downloading the blockchain and connecting to peers. However, if you want to help the network and have your node start validating transactions, you'll need to open a port for incoming internet traffic. This may require changing the settings of your router (to allow outbound connections). If this is the case:
		1. Find your [Local IP address](https://www.howtogeek.com/236838/how-to-find-any-devices-ip-address-mac-address-and-other-network-connection-details/) and [External IP address](https://whatismyipaddress.com/).
		1. Edit the settings of your router to open port 9108 (port forwarding). This [process](https://www.wikihow.com/Set-Up-Port-Forwarding-on-a-Router) depends on the type of router you have. 
	- If running a firewall, you'll also need to configure it to allow inbound connections on port 9108.

Reply:
> However, if you want to help the network and have your node start validating transactions, you'll need to open a port for incoming internet traffic.

Opening the port is not required for validating transactions. I would suggest something like the following:

> To maximise your positive impact on the Decred network, it is important to allow inbound peers to connect to your node. Allowing inbound peers will allow new participants in the Decred network to connect to your node, and also enable your node to serve lightweight clients such as SPV wallets.

I found some nice reading on this topic - the first post from David in this forum thread
https://forum.decred.org/threads/how-dcrd-works-and-the-shortage-of-public-dcrd-nodes.1371/

Its a very good thread, lots in there which could be taken for this new article you are writing.

Reply:
Have gone with your suggested text. 

Wondering if it makes sense to add material from that [forum post](https://forum.decred.org/threads/how-dcrd-works-and-the-shortage-of-public-dcrd-nodes.1371/). We could maybe add finding your IP address or verifying inbound connection by looking for,
```[INF] BMGR: New valid peer 115.66.31.148:52400 (inbound) (/dcrwire:0.0.1/dcrd:0.0.8/)```

Too low level? Outdated?

Reply:
Good idea, we could keep it short and sweet just with something like:

> You will know when your node is correctly configured because you will see inbound peers in the logs. Inbound peers can also be checked by using the RPC `getpeerinfo`

Reply:
Have added bullet:

>- If you have configured your node to allow inbound peers, you should see inbound peers in the logs. Inbound peers can also be checked using the Remote Procedure Call (RPC) command [getpeerinfo](../wallets/cli/dcrctl-rpc-commands.md).